### DISCUSSION TITLE
Fail to load URDF

### DISCUSSION THREAD

**Discussion by KingWang569** [foxglove/community#646] [Q&A] [answered]
I have successfully connected Foxglove and my ROS host, and can visualize many topics, but when I try to display the URDF model in real time, I encountered a problem:
1. I try to add URDF in the Panel Setting of 3D, Custom Layers, but I don’t know how to fill in the URL of the URDF, how to fill in the local host and the remote host respectively
2. I tried to directly drag and drop URDF (or XACRO) files into 3D Pannel, and various errors were reported. These URDF (or XACRO) can be displayed normally through Rviz, but Foxglove will prompt that STL, dae, etc. cannot be loaded

- Version: latest web app
- Platform: Win10/ubuntu18.04

I am very eager for your help, is there a demo for loading URDF (XACRO)？

![img_v2_d0c1bc1e-9081-426f-8941-fb20dbd89f9g](https://user-images.githubusercontent.com/96674018/221401032-e19d6ea1-e11f-4751-8b1c-0d264dffc382.png)
![img_v2_d9a08446-c24e-480c-9dad-cc09fffed71g](https://user-images.githubusercontent.com/96674018/221401037-c876481d-8151-4488-9225-442ad5eb7912.png)
![Htqba5G35k](https://user-images.githubusercontent.com/96674018/221401043-cd6719bd-e306-4d89-aadb-d89a437b639f.jpg)

**Comment by foxhubber** [comment]
Linear: [FG-2184](https://linear.app/foxglove/issue/FG-2184/fail-to-load-urdf-[foxglovestudio-5433])

**Comment by jtbandes** [comment, marked as answer]
Using the web app requires `http://`/`https://` URLs (or relative urls for meshes). For the desktop app you can also use `file://` URLs or `package://`. Using `package://` requires your ROS_PACKAGE_PATH to be configured. The app can get this from an environment variable or you can configure it in the ⚙️ Settings pane.

Drag & drop of URDF assets is related to the deprecated URDF Viewer panel and will be removed soon. It's recommended that you use the 3D panel settings instead: https://foxglove.dev/docs/studio/panels/3d#add-urdf
